winpr/libwinpr/utils/wlog/wlog.c

```diff
@@ -302,6 +305,92 @@ DWORD WLog_GetLogLevel(wLog* log)
 	}
 }
 
+BOOL WLog_SetStringLogLevel(wLog* log, LPCSTR level)
+{
+	int lvl;
+	if (!log || !level)
+		return FALSE;
+
+	lvl = WLog_ParseLogLevel(level);
+	if (lvl < 0)
+		return FALSE;
+
+	return WLog_SetLogLevel(log, lvl);
+}
+
+BOOL WLog_AddStringLogFilters(LPCSTR filter)
+{
+	DWORD pos;
+	DWORD size;
+	DWORD count;
+	DWORD status;
+	LPSTR p;
+	LPSTR s;
+	LPSTR cp;
+	wLogFilter* tmp;
+
+	if (!filter)
+		return FALSE;
+
+	count = 1;
+	p = (LPSTR)filter;
+
+	while ((p = strchr(p, ',')) != NULL)
+	{
+		count++;
+		p++;
+	}
+
+	pos = g_FilterCount;
+	size = g_FilterCount + count;
+	tmp = (wLogFilter*) realloc(g_Filters, size * sizeof(wLogFilter));
+
+	if (!tmp)
+	{
+		free (g_Filters);
+		return FALSE;
+	}
+	g_Filters = tmp;
+
+	cp = (LPSTR)_strdup(filter);
+	if (!cp)
+		return FALSE;
+
+	p = cp;
+	s = cp;
+
+	do
+	{
+		p = strchr(p, ',');
```

Comment:
isn't it 
p = strchr(s, ','); 
here ?

May I suggest renaming s to startAt and p to commaPos ?